app/client/src/components/RoleGuard.tsx

```diff
@@ -0,0 +1,14 @@
+import router from 'next/router';
+
+export interface RoleGuardProps {
+    organizer?: boolean;
+    moderator?: boolean;
+    children: JSX.Element | JSX.Element[];
+}
+
+export function RoleGuard({ organizer = false, moderator = false, children }: RoleGuardProps){
+    if (organizer) return <>{children}</>
+    else router.push('/dashboard');
+    if (moderator) return <>{children}</>
```

Comment:
This could be simplified.

```jsx
if (organizer || moderator) return <>{children}</>
return <></>
```

Although, it may be more complicated depending on what you wanna do per my comment further up in this file.